File: local-movie-app/src/main/java/com/github/rahmnathan/localmovies/app/control/MediaPathUtils.java
package com.github.rahmnathan.localmovies.app.control;

import java.io.File;
import java.util.Arrays;
import java.util.stream.Collectors;

class MediaPathUtils {

    static String getParentPath(String path){
        String[] dirs = path.split(File.separator);
        return Arrays.stream(dirs)
                .limit(dirs.length - 1)
                .collect(Collectors.joining(File.separator));
    }

    static String getFilename(String path){
        String[] directoryList = path.split(File.separator);
        return directoryList[directoryList.length - 1];
    }
}

File: local-movie-app/src/main/java/com/github/rahmnathan/localmovies/app/media/provider/control/MediaPathUtils.kt
package com.github.rahmnathan.localmovies.app.media.provider.control

object MediaPathUtils {
    fun getParentPath(path: String): String {
        return path.substringBeforeLast('/', "")
    }

    fun getFilename(path: String): String {
        return path.substringAfterLast('/')
    }
}
